lib/common/xml.c

```diff
@@ -3786,9 +3786,14 @@ dump_xml_element(xmlNode * data, int options, char **buffer, int *offset, int *m
 
     if (data->children) {
         xmlNode *xChild = NULL;
-
-        for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
-            crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
+        if (options & xml_log_option_text) {
+            for(xChild = data->children; xChild != NULL; xChild = xChild->next) {
+                crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
+            }
+        } else {
+            for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
+                crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
+            }
```

Comment:
it is right if the functions __xml_first_child and __xml_next do not change, but if they change, there may be difference.